### PR TITLE
Fixed #28248 -- Fixed password reset tokens being valid for 1 day longer than PASSWORD_RESET_TIMEOUT_DAYS.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/28248

### CODE REVIEW

**Comment by spookylukey** [comment]
I believe this change is incorrect and should be reverted, according to a common sense interpretation of the setting.

The code rounds all timestamps to midnight (server time) providing a resolution of only 1 day. So if you generate a links 5 minutes before midnight and try to use it 6 minutes later, that counts as 1 day. With the current code, if you set the timeout setting to 1 day, such a link would be rejected.

In other words, new code interprets timeout of 1 day as "up to one day, could be almost zero". The old way interpreted as "at least 1day, could be up to 2". I think the old way was better, much less likely to be an accidental usability issue. There is virtually no security concern with being too generous here (see my latest post on django-devs on related issue).
